fsck/fsck.c

```diff
@@ -557,7 +557,7 @@ static int resolve_path(struct path_resolve_ctx *ctx, struct exfat_inode *child)
 		memcpy((char *)utf16_path, (char *)ctx->ancestors[i]->name,
 				name_len * 2);
 		utf16_path += name_len;
-		memcpy((char *)utf16_path, u"/", 2);
```

Comment:
Could you explain why you remove the prefix u?

Reply:
LLVM didn't recognize it.
Maybe there is a missing include?

Reply:
I think this u standard for unsigned?
If that's the case it shouldn't matter

Reply:
`u` is a prefix for composing string of C11 type `char16_t`. Maybe you have tried to compiled it with turned C11 support off?

Chaning to `memcpy((char *)utf16_path, "/", 2);` is a wrong as depends on endianity.

Reply:
Oh I haven't thought about endianness.
I'm surely building with the latest C version so that's likely the case.

Anyways I see you now replaced the method, thanks.

Reply:
> I'm surely building with the latest C version so that's likely the case.

IIRC last standard is C17 which is just fixup of C11. So it looks like your compiler does not have (full) support for last C standard if you get error for `u`-strings.

> Anyways I see you now replaced the method, thanks.

Yes, I replaced it by composing __le16 slash manually in variable. That should work also with old ANSI C89, just needs one additional variable.